### PR TITLE
fix-NXP-26655-CollectionIdsNotResolved

### PR DESCRIPTION
**Pull request by nmpcunha**
https://qa.nuxeo.org/jenkins/view/TestAndPush/job/TestAndPush/job/ondemand-testandpush-ncunha/23/

### CODE REVIEW

**Comment by guirenard** [comment]
:+1:

File: nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/model/DocumentModelResolver.java [hunk 1]
```diff
@@ -168,13 +168,16 @@ public boolean validate(Object value) {
...
     public boolean validate(Object value, Object context) {
-        MutableBoolean validated = new MutableBoolean();
-        resolve(value, context, (session, docRef) -> {
-            if (session.exists(docRef)) {
-                validated.setTrue();
-            }
-        });
-        return validated.isTrue();
+        if (validation) {
+            MutableBoolean validated = new MutableBoolean();
```
Comment (efge) [inline nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/model/DocumentModelResolver.java]: Please don't introduce extra indentation when you can just do early exit:

    if (!validation) {
        return true;
    }
    // original code here

This also makes the diff easier to read.

**Comment by efge** [comment]
Please create a separate ticket just for the first commit, which has nothing to do with collections, it's about the `DocumentModelResolver` not taking into account the validation flag correctly.

**Comment by guirenard** [comment]
:+1:

**Comment by Gabez0r** [comment]
👍

**Pull request merged**